### ISSUE TITLE
Server Crash: Magnify and Realm

### ISSUE THREAD

**Issue by pauldpickell** [ubc-minetest-classroom/Minetest_Classroom#121]
**Describe the bug**
Start Minetest_Classroom gamdeid on server and get the following output on debug:

```
2022-07-12 19:14:55: ACTION[Main]: Server: Shutting down
2022-07-12 19:14:55: ERROR[Main]: ModMetadataDatabaseFiles[magnify]: failed to write file.
2022-07-12 19:14:55: ERROR[Main]: ModError: Mod security: Blocked attempted read from /home/ubuntu/minetest/bin/../worlds/ubc_campus_mosaic\realmPrivilegeWhitelist.conf
```

**To Reproduce**
1. Run server command `./minetestserver --gameid Minetest_Classroom --worldname ubc_campus_mosaic`

**Expected behavior**
Server should start and load defined world.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04.6 LTS

**Comment by lukasgolson** [comment]
Looking into this right now

**Comment by Lowie375** [comment]
Will look into why `magnify` was unable to write to its database

UPDATE: It seems like that may have been an issue on MineTest's end, since the error is coming from [a function which handles saving data to databases within the game](https://github.com/minetest/minetest/blob/master/src/database/database-files.cpp#L452-L457). Unsure of the specifics causing the write to fail, since the issue appears to come from the file system itself

**Comment by lukasgolson** [comment]
I have a sus

> Will look into why `magnify` was unable to write to its database
> 
> UPDATE: It seems like that may have been an issue on MineTest's end, since the error is coming from [a function which handles saving data to databases within the game](https://github.com/minetest/minetest/blob/master/src/database/database-files.cpp#L452-L457). Unsure of the specifics causing the write to fail, since the issue appears to come from the file system itself

For Realms, I have a suspicion it's with the filepath. I'm still updating ubuntu packages to test my idea. The path that is being used for the file might invalid on Ubuntu but valid on Windows due to the way that filepaths are handled. 

In the Windows file API, the backslash is interpreted the same as a forward slash. I'm not sure if this is the same as Ubuntu.

In terms of debugging, I am able to run the server with no issues on singleplayer and as a server. I need to test if it has anything to do with specifying world paths, however.

**Comment by pauldpickell** [comment]
Okay, I have solved the magnify error. Need to run `minetestserver` with `sudo`. The other error still persists.

**Comment by Lowie375** [comment]
> The path that is being used for the file might invalid on Ubuntu but valid on Windows due to the way that filepaths are handled.

This may not have been the issue this time around, but it does raise a valid concern. We may want to look into how file paths are handled on different operating systems and set a standard slash to use across the MineTest Classroom mods for consistency and cross-system compatibility

**Comment by lukasgolson** [comment]
Tentative fix in #122 running on Minetest 5.5.1 built with LuaJIT from sources on Ubuntu 20.04; will test with production configurations.

**Comment by pauldpickell** [comment]
Fixed by #122
